foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/platform/database/PostgreSQL10Platform.java

```diff
@@ -0,0 +1,198 @@
+    ) throws SQLException {
+        if (PG_OBJECT_CLASS == null) {
+            throw new PersistenceException("Class org.postgresql.util.PGobject was not found");
+        }
+        if (PG_OBJECT_CLASS.isInstance(parameter)) {
+            statement.setObject(name, parameter);
+        } else {
+            super.setParameterValueInDatabaseCall(parameter, statement, name, session);
+        }
+    }
+
+    // Postgres specific JSON types support:
+    // Stores JsonValue instances as JSONB.
+    /**
+     * INTERNAL:
+     * Convert JSON value field to JDBC statement type.
+     * Postgres JSON storage type is {@code JSONB} and target Java type is {@code PGobject}.
+     *
+     * @param <T> classification type
+     * @param jsonValue source JSON value field
+     * @return converted JDBC statement type
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T convertJsonValueToDataValue(final JsonValue jsonValue) throws PersistenceException {
+        if (PG_OBJECT_CLASS == null) {
+            throw new PersistenceException("Class org.postgresql.util.PGobject was not found");
+        }
+        final String jsonAsString = super.convertJsonValueToDataValue(jsonValue);
+        // Following code is called trough reflection to avoid org.postgresql.util.PGobject dependency
+        //        PGobject pgObject = new PGobject();
+        //        pgObject.setValue(jsonAsString); //json string
+        //        pgObject.setType("jsonb");
+        try {
+            Constructor<?> pgConstructor = PG_OBJECT_CLASS.getConstructor();
+            Object pgObject = pgConstructor.newInstance();
+            Method setValue = PG_OBJECT_CLASS.getDeclaredMethod("setValue", String.class);
+            Method setType = PG_OBJECT_CLASS.getDeclaredMethod("setType", String.class);
+            setValue.invoke(pgObject, jsonAsString);
+            setType.invoke(pgObject, "jsonb");
+            return (T) pgObject;
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new PersistenceException("JSON value to database type conversion failed.", e);
+        }
+    }
+
+    /**
+     * Retrieve JSON data from JDBC {@code ResultSet}.
+     * JSON data retrieved from Postgres JDBC {@code ResultSet} are returned as {@code PGobject} instance.
+     * It must be converted to {@code String} first to be accepted by common {@code JsonTypeConverter}.
+     *
+     * @param resultSet source JDBC {@code ResultSet}
+     * @param columnNumber index of column in JDBC {@code ResultSet}
+     * @return JSON data from JDBC {@code ResultSet} as {@code String} to be parsed by common {@code JsonTypeConverter}
+     * @throws SQLException if data could not be retrieved
+     */
+    @Override
+    public Object getJsonDataFromResultSet(ResultSet resultSet, int columnNumber) throws SQLException {
```

Comment:
This method working with `PGobject` would be nice without reflection. But I'm wondering that it may be performance regression compared to default code as it is now.
Default getJsonDataFromResultSet code with `resultSet.getString(columnNumber);` works too with Postgres so maybe I can delete this method.